### PR TITLE
Adds setting to configure leaving stream open

### PR DESCRIPTION
**Pull request by irvinesunday**
Fixes https://github.com/microsoft/OpenAPI.NET/issues/530
Closes https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/583

This PR:
- Adds a new `boolean` setting in `OpenApiReaderSettings.cs`: `LeaveStreamOpen` to flag whether or not to leave the stream object open after reading from an `OpenApiStreamReader` object.
Creating a `StreamReader` object within a `using` statement closes the `stream` when the `Dispose()` method of the `StreamReader` is called.
- Creates `OpenApiStreamReader` tests to validate setting the `LeaveStreamOpen` setting to either `true` or `false`.

### CODE REVIEW

File: src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs [hunk 1]
```diff
@@ -29,13 +29,20 @@ public OpenApiStreamReader(OpenApiReaderSettings settings = null)
...
         /// </summary>
         /// <param name="input">Stream containing OpenAPI description to parse.</param>
-        /// <param name="diagnostic">Returns diagnostic object containing errors detected during parsing</param>
-        /// <returns>Instance of newly created OpenApiDocument</returns>
+        /// <param name="diagnostic">Returns diagnostic object containing errors detected during parsing.</param>
+        /// <returns>Instance of newly created OpenApiDocument.</returns>
         public OpenApiDocument Read(Stream input, out OpenApiDiagnostic diagnostic)
         {
-            using (var reader = new StreamReader(input))
+            if (_settings.LeaveStreamOpen)
```
Comment (baywet) [inline src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs:36]:
```CSharp
var reader = new StreamReader(input);
var result = new OpenApiTextReaderReader(_settings).Read(reader, out diagnostic);
if (!_settings.LeaveStreamOpen)
   reader.Dispose();
return result;
```

File: src/Microsoft.OpenApi.Readers/OpenApiReaderSettings.cs [hunk 2]
```diff
@@ -61,11 +61,17 @@ public class OpenApiReaderSettings
...
         /// <remarks>
         /// Default loader will attempt to dereference http(s) urls and file urls.
         /// </remarks>
         public IStreamLoader CustomExternalLoader { get; set; }
+
+        /// <summary>
+        /// Whether to leave the <see cref="Stream"/> object open after reading
+        /// from an <see cref="OpenApiStreamReader"/> object.
+        /// </summary>
+        public bool LeaveStreamOpen { get; set; } = false;
```
Comment (baywet) [inline src/Microsoft.OpenApi.Readers/OpenApiReaderSettings.cs:75]:
```suggestion
        public bool KeepStreamOpen { get; set; };
```
boolean is always false by default.
nit on naming

**Review by baywet — CHANGES_REQUESTED** [review]

File: src/Microsoft.OpenApi.Readers/OpenApiReaderSettings.cs (hunk 2 above)
Comment (irvinesunday) [reply src/Microsoft.OpenApi.Readers/OpenApiReaderSettings.cs:75]: 👍🏼 on the boolean default.

On the naming: I wanted to remain as close to the framework naming as possible: See: https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.-ctor?view=netcore-3.1#System_IO_StreamReader__ctor_System_IO_Stream_System_Text_Encoding_System_Boolean_System_Int32_System_Boolean_
image

File: src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs (hunk 1 above)
Comment (irvinesunday) [reply src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs:36]: Yes. Much cleaner!

Comment (irvinesunday) [reply src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs:36]: I like brackets after `if` statements though, even for one-liners 😄

File: src/Microsoft.OpenApi.Readers/OpenApiReaderSettings.cs (hunk 2 above)
Comment (baywet) [reply src/Microsoft.OpenApi.Readers/OpenApiReaderSettings.cs:75]: perfect, ignore my naming suggestion then. Thanks for doing the research

File: src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs (hunk 1 above)
Comment (baywet) [reply src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs:36]: I won't fight coding preferences. It seems that I'm the only one on the team to like one liners with no brackets 😅

**Review by darrelmiller — APPROVED** [review]

**Pull request merged**